src/Compilers/Core/Portable/ConstantValueSpecialized.cs

```diff
@@ -201,6 +210,21 @@ internal override string GetValueToDisplay()
             {
                 return (_value == null) ? "null" : string.Format("\"{0}\"", _value);
             }
+
+            public override string ToString(string? format, IFormatProvider? provider)
+            {
+                int formatLength = RopeValue.Length;
+                if (format is not null && int.TryParse(format, out var len) &&
+                    len >= 0)
+                {
+                    formatLength = len;
+                }
+
+                bool trim = formatLength >= 3 && formatLength < RopeValue.Length;
```

Comment:
>formatLength >= 3 [](http://example.com/codeflow?start=28&length=17)

I don't know if it's worth subtracting `3` from the length (for the `"`, and `..."`?) since it looks like we're not counting the `"` when `formatLength >= RopeValue.Length`. 

Perhaps just assume the format length is for the string content only, not the prefix and suffix.